crates/services/gas_price_service/src/v1/uninitialized_task.rs

```diff
@@ -0,0 +1,383 @@
+use crate::{
+    common::{
+        fuel_core_storage_adapter::{
+            block_bytes,
+            get_block_info,
+            mint_values,
+            GasPriceSettings,
+            GasPriceSettingsProvider,
+        },
+        gas_price_algorithm::SharedGasPriceAlgo,
+        l2_block_source::FuelL2BlockSource,
+        updater_metadata::UpdaterMetadata,
+        utils::{
+            BlockInfo,
+            Error as GasPriceError,
+            Result as GasPriceResult,
+        },
+    },
+    ports::{
+        GasPriceData,
+        GasPriceServiceConfig,
+        L2Data,
+        MetadataStorage,
+    },
+    v1::{
+        algorithm::SharedV1Algorithm,
+        da_source_service::service::{
+            DaBlockCostsSource,
+            DaSourceService,
+        },
+        metadata::{
+            v1_algorithm_from_metadata,
+            V1AlgorithmConfig,
+            V1Metadata,
+        },
+        service::GasPriceServiceV1,
+    },
+};
+use anyhow::Error;
+use fuel_core_services::{
+    stream::BoxStream,
+    RunnableService,
+    ServiceRunner,
+    StateWatcher,
+};
+use fuel_core_storage::{
+    not_found,
+    transactional::AtomicView,
+};
+use fuel_core_types::{
+    fuel_tx::field::MintAmount,
+    fuel_types::BlockHeight,
+    services::block_importer::SharedImportResult,
+};
+use fuel_gas_price_algorithm::v1::AlgorithmUpdaterV1;
+
+pub struct UninitializedTask<
```

Comment:
I'm not following why we need a separate `UninitializedTask`. As I read this, the purpose of this type is to separate task creation (`UninitializedTask::new`) and initialization `UninitializedTask::init`. I don't see any extra information being provided during initialization, so what's the benefit of doing this over having a single function that returns `GasPriceServiceV1` directly?

Reply:
Summary: The initialization logic depends on database concepts (getting blocks at different heights etc.) that we don't want the gas price algo to be aware of, to maintain a good separation between domain and integration logic. Ideally this initialization logic shouldn't even be here but rather closer to the fuel-core code that starts the services, but for the time being it's better to keep it isolated to this particular task.